framework/bootstrap.go

```diff
@@ -64,7 +62,10 @@ func (f *framework) Start() {
 	f.task = f.taskBuilder.GetTask(f.taskID)
 	f.topology.SetTaskID(f.taskID)
 
-	go f.startHTTP()
+	// There is a possible race:
+	// A task restarts and has nothing. But a fetch still comes.
+	// How to fix? Use node to node channel (pipeline) instead of single request every time.
```

Comment:
Make sense. It should be the next goal we have in refactor work since this patch is too big.

I will change the comments.